### DISCUSSION TITLE
Sonoff SV won’t exit initial setup mode, can’t add to Home app

### DISCUSSION THREAD

**Discussion by creatoro** [RavenSystem/esp-homekit-devices#1614] [Q&A]
I’ve been trying to get a Sonoff SV work as a switch using the sample from the Wiki.

I’ve got to a point where after the initial setup the device will connect to the WiFi network and can be reached via 192.168.1.x:4567.

However, it won’t get past the initial setup page (orange background) and therefore I can’t add it to the Home app. Also, the pre-configured host HAA-XXXYYY:4567 is unreachable.

As I understand after the initial setup the setup page should be a white background page with on option to reset HomeKit ID. This is not what I get.

Please help with solving this situation.

**Comment by tonysprenk** [comment]
Perhaps your json is invalid. Can you post it here?

**Reply by creatoro** [reply]
Sure:

{
  "c": { "l": 13, "b": [{ "g": 0, "t": 5 }] },
  "a": [{
    "t": 1,
    "d": 3600,
    "s": 5,
    "0": { "r": [{ "g": 12, "v": 0 }] },
    "1": { "r": [{ "g": 12, "v": 1 }] },
    "b": [{ "g": 0 }]
  }]
}

**Comment by tonysprenk** [comment]
The json is valid. After you select the save button in the initial setup screen the device will download the latest version of haa this can take about 7 minutes. Have you waited before trying to add to HomeKit or accessing the setup page again?

**Reply by creatoro** [reply]
This is what I do during the install process:

1. Hold down button while connecting to USB
2. Erase flash
3. Disconnect USB and repeat step 1.
4. Write fullhaaboot to Sonoff SV
5. Disconnect USB and reconnect
6. Connect to Sonoff SV via Wifi to enter setup using 192.168.4.1:4567
7. Enter JSON string, select SSID and enter Wifi password, then click Save
8. Wait 10 minutes
9. Disconnect Sonoff SV from USB and reconnect
10. Try to add to Home using QR code

Nothing happens when I try to add to Home app.

As I understand I should get a notification about an unsigned or unofficial HomeKit device as soon as I try to add the device. This doesn’t happen either…

**Reply by creatoro** [reply]
Another strange phenomena: I can reach the device via 192.168.1.x:4567 (the page that should be accessible after initial setup) and at the same time the SSID “HAA-XXXYYY” is still being broadcasted (I think this should be unavailable after the initial setup is complete).

I can reach the setup page on both addresses: 192.168.1.x:4567 and 192.168.4.1:4567

It’s like the device is stuck between the initial setup and the completed configuration somewhere.

**Reply by Ucoz3000** [reply]
> Another strange phenomena: I can reach the device via 192.168.1.x:4567 (the page that should be accessible after initial setup) and at the same time the SSID “HAA-XXXYYY” is still being broadcasted (I think this should be unavailable after the initial setup is complete).
> 
> I can reach the setup page on both addresses: 192.168.1.x:4567 and 192.168.4.1:4567
> 
> It’s like the device is stuck between the initial setup and the completed configuration somewhere.

with firmware 9.0.4, the same strange phenomena appeared. I tried to flash 9.0.0-9.0.4, the result was unsuccessful.

**Reply by Gejza75** [reply]
Hi, I have the same problem. Previously working well without any problem. After power outage fails to start. No flashing light. After reflashing with any FW (even old one and tried also newest one) stays as you described. Somewhere in the middle. First I thought that there is some hardware issue, but after reflashing Tasmota FW everything was working well. But I do not want Tasmota... I wanted to have it in my Homekit environment.. This is the script:

{
  "c": {
    "l": 13,
    "b": [
      {
        "g": 0,
        "t": 5
      }
    ]
  },
  "a": [
    {
      "0": {
        "r": [
          {
            "g": 12
          }
        ]
      },
      "1": {
        "r": [
          {
            "g": 12,
            "v": 1
          }
        ]
      },
      "b": [
        {
          "g": 0,
          "t": 2,
          "p": 0,
          "i": 0
        }
      ],
      "s": 0,
      "i": 0.2
    }
  ]
}


Just opening garage door... Simple. Any idea?

**Comment by didl001** [comment]
I have the same problem, after power loss the Sonoff no longer react or any light come on. 
I flashed another one with the same Firmware always working, but it stuck in the Config, after setting the JSON (Valid one always work before) and get in the Wifi password, the Sonoff won't get out of the process and always keep his own Wifi visible.
Also don't show up after 30 mins of waiting on the Home App.
Anyone have solved this problem?

**Reply by creatoro** [reply]
For me, this turned out to be a network issue. There was a router with incorrect config on the network which made things very messy. When the router was removed, everything worked as it should.

**Reply by didl001** [reply]
Okay thx, got to try it. But all other HAA devices still work and also nothing in my Network were changed, so I wonder.

**Reply by didl001** [reply]
Don't even work on a Factory reseter router, always keep stuck in the config and always keeps showing the own made Network
